src/main/java/com/celements/search/web/WebSearchService.java

```diff
@@ -0,0 +1,119 @@
+import com.celements.rights.access.EAccessLevel;
+import com.celements.rights.access.IRightsAccessFacadeRole;
+import com.celements.search.lucene.ILuceneSearchService;
+import com.celements.search.lucene.LuceneSearchResult;
+import com.celements.search.lucene.query.LuceneQuery;
+import com.celements.search.web.classes.WebSearchConfigClass;
+import com.celements.search.web.packages.WebSearchPackage;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.web.Utils;
+
+@Component
+public class WebSearchService implements IWebSearchService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WebSearchService.class);
+
+  @Requirement
+  private IModelAccessFacade modelAccess;
+
+  @Requirement
+  private IRightsAccessFacadeRole rightsAccess;
+
+  @Requirement
+  private ILuceneSearchService luceneSearchService;
+
+  @Requirement
+  private List<WebSearchPackage> webSearchPackages;
+
+  @Requirement
+  private ModelContext context;
+
+  @Override
+  public Set<WebSearchPackage> getAvailablePackages(DocumentReference configDocRef) {
+    Set<WebSearchPackage> searchPackages = new HashSet<>();
+    try {
+      XWikiDocument configDoc = modelAccess.getDocument(configDocRef);
+      searchPackages.addAll(getConfiguredPackages(configDoc));
+      if (searchPackages.isEmpty()) {
+        searchPackages.addAll(getDefaultPackages());
+      }
+      searchPackages.addAll(getRequiredPackages(configDocRef));
+    } catch (DocumentNotExistsException exp) {
+      LOGGER.error("Failed to load configDoc '{}'", configDocRef, exp);
+    }
+    return searchPackages;
+  }
+
+  private List<WebSearchPackage> getConfiguredPackages(XWikiDocument configDoc)
+      throws DocumentNotExistsException {
+    return modelAccess.getFieldValue(configDoc, WebSearchConfigClass.FIELD_PACKAGES).or(
+        Collections.<WebSearchPackage>emptyList());
+  }
+
+  private List<WebSearchPackage> getDefaultPackages() {
+    return FluentIterable.from(webSearchPackages).filter(
+        WebSearchPackage.PREDICATE_DEFAULT).toList();
+  }
+
+  private List<WebSearchPackage> getRequiredPackages(DocumentReference configDocRef)
+      throws DocumentNotExistsException {
+    final XWikiDocument configDoc = modelAccess.getDocument(configDocRef);
+
+    return FluentIterable.from(webSearchPackages).filter(new Predicate<WebSearchPackage>() {
+
+      @Override
+      public boolean apply(WebSearchPackage searchPackage) {
+        return searchPackage.isRequired(configDoc);
+      }
+    }).toList();
+  }
+
+  @Override
+  public WebSearchQueryBuilder createWebSearchBuilder(DocumentReference configDocRef)
+      throws DocumentNotExistsException {
+    WebSearchQueryBuilder ret = null;
+    ret = Utils.getComponent(WebSearchQueryBuilder.class);
+    if ((configDocRef != null) && rightsAccess.hasAccessLevel(configDocRef, EAccessLevel.VIEW)) {
```

Comment:
move the rights check to WebSeachScriptService since it's meant as security for velocity calls. it has to happen in webSearch(...) as well as in createWebSearchBuilder(...)